From 5da64d8784d36c0601743a5159a598f5888089c7 Mon Sep 17 00:00:00 2001
From: Gibeom Kim <gibeomii.kim@samsung.com>
Date: Thu, 1 Apr 2021 17:52:46 +0900
Subject: [PATCH] cifsd: remove stale prototype and variables
Git-commit: 5da64d8784d36c0601743a5159a598f5888089c7
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


Remove unused function prototype and variables.

Signed-off-by: Gibeom Kim <gibeomii.kim@samsung.com>
Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/cifsd/glob.h    | 1 -
 fs/cifsd/oplock.h  | 3 ---
 fs/cifsd/server.h  | 2 --
 fs/cifsd/smbacl.h  | 1 -
 fs/cifsd/unicode.h | 2 --
 fs/cifsd/vfs.h     | 3 ---
 6 files changed, 12 deletions(-)

diff --git a/fs/cifsd/glob.h b/fs/cifsd/glob.h
index 27500afbeaf5..d0bc6edd0477 100644
--- a/fs/cifsd/glob.h
+++ b/fs/cifsd/glob.h
@@ -19,7 +19,6 @@
 /* @FIXME clean up this code */
 
 extern int ksmbd_debug_types;
-extern int ksmbd_caseless_search;
 
 #define DATA_STREAM	1
 #define DIR_STREAM	2
diff --git a/fs/cifsd/oplock.h b/fs/cifsd/oplock.h
index 5b6615f99c76..f8b4b486eb93 100644
--- a/fs/cifsd/oplock.h
+++ b/fs/cifsd/oplock.h
@@ -123,9 +123,6 @@ void create_mxac_rsp_buf(char *cc, int maximal_access);
 void create_disk_id_rsp_buf(char *cc, __u64 file_id, __u64 vol_id);
 void create_posix_rsp_buf(char *cc, struct ksmbd_file *fp);
 struct create_context *smb2_find_context_vals(void *open_req, const char *str);
-int ksmbd_durable_verify_and_del_oplock(struct ksmbd_session *curr_sess,
-		struct ksmbd_session *prev_sess, int fid, struct file **filp,
-		u64 sess_id);
 struct oplock_info *lookup_lease_in_table(struct ksmbd_conn *conn,
 		char *lease_key);
 int find_same_lease_key(struct ksmbd_session *sess, struct ksmbd_inode *ci,
diff --git a/fs/cifsd/server.h b/fs/cifsd/server.h
index 7b2f6318fcff..b682d28963e8 100644
--- a/fs/cifsd/server.h
+++ b/fs/cifsd/server.h
@@ -17,8 +17,6 @@
 #define SERVER_CONF_SERVER_STRING	1
 #define SERVER_CONF_WORK_GROUP		2
 
-extern int ksmbd_debugging;
-
 struct ksmbd_server_config {
 	unsigned int		flags;
 	unsigned int		state;
diff --git a/fs/cifsd/smbacl.h b/fs/cifsd/smbacl.h
index 9b22bff4191f..032b6a3ec6f4 100644
--- a/fs/cifsd/smbacl.h
+++ b/fs/cifsd/smbacl.h
@@ -193,7 +193,6 @@ int smb_inherit_dacl(struct ksmbd_conn *conn, struct dentry *dentry,
 		unsigned int uid, unsigned int gid);
 int smb_check_perm_dacl(struct ksmbd_conn *conn, struct dentry *dentry,
 		__le32 *pdaccess, int uid);
-int store_init_posix_acl(struct inode *inode, umode_t perm);
 int set_info_sec(struct ksmbd_conn *conn, struct ksmbd_tree_connect *tcon,
 		struct dentry *dentry, struct smb_ntsd *pntsd, int ntsd_len,
 		bool type_check);
diff --git a/fs/cifsd/unicode.h b/fs/cifsd/unicode.h
index c37d7024cf60..68f1c8290911 100644
--- a/fs/cifsd/unicode.h
+++ b/fs/cifsd/unicode.h
@@ -71,8 +71,6 @@ int smbConvertToUTF16(__le16 *target, const char *source, int srclen,
 char *ksmbd_extract_sharename(char *treename);
 #endif
 
-wchar_t cifs_toupper(wchar_t in);
-
 /*
  * UniStrcat:  Concatenate the second string to the first
  *
diff --git a/fs/cifsd/vfs.h b/fs/cifsd/vfs.h
index e1ca9ac11ba5..b41b23d40636 100644
--- a/fs/cifsd/vfs.h
+++ b/fs/cifsd/vfs.h
@@ -205,8 +205,6 @@ int ksmbd_vfs_remove_file(struct ksmbd_work *work, char *name);
 int ksmbd_vfs_link(struct ksmbd_work *work,
 		const char *oldname, const char *newname);
 int ksmbd_vfs_getattr(struct path *path, struct kstat *stat);
-int ksmbd_vfs_symlink(const char *name, const char *symname);
-int ksmbd_vfs_readlink(struct path *path, char *buf, int lenp);
 
 int ksmbd_vfs_fp_rename(struct ksmbd_work *work, struct ksmbd_file *fp,
 		char *newname);
@@ -230,7 +228,6 @@ int ksmbd_vfs_setxattr(struct dentry *dentry, const char *attr_name,
 		const void *attr_value, size_t attr_size, int flags);
 int ksmbd_vfs_xattr_stream_name(char *stream_name, char **xattr_stream_name,
 		size_t *xattr_stream_name_size, int s_type);
-int ksmbd_vfs_truncate_xattr(struct dentry *dentry, int wo_streams);
 int ksmbd_vfs_remove_xattr(struct dentry *dentry, char *attr_name);
 void ksmbd_vfs_xattr_free(char *xattr);
 int ksmbd_vfs_kern_path(char *name, unsigned int flags, struct path *path,
-- 
2.33.0

